### PR TITLE
[DCOS-37811] Fix Kafka cipher test flakes

### PR DESCRIPTION
**Pull request by mpereira**
- Don't blow up if we don't have the RFC name for an OpenSSL cipher.
- Warn us when OpenSSL includes new ciphers.

### CODE REVIEW

File: testing/security/cipher_suites.py [hunk 1]
```diff
@@ -262,9 +262,15 @@
 RFC_TO_OPENSSL_NAMES = sdk_utils.invert_dict(OPENSSL_TO_RFC_NAMES)
 
 
+def missing_ciphers(openssl_ciphers: set) -> bool:
```
Comment (elezar) [inline testing/security/cipher_suites.py]: The return value is also a set and not a bool.

**Comment by mpereira** [comment]
With these changes:
- Tests won't break if new ciphers are added to OpenSSL
- Tests will output the list of ciphers we're missing from our list, so we can add them at our convenience

File: frameworks/kafka/tests/test_tls.py [hunk 2]
```diff
@@ -120,24 +122,33 @@ def test_tls_ciphers(kafka_service_tls):
...
         json=True), '').rstrip().split(','))
-    possible_ciphers = set(map(cipher_suites.rfc_name, sdk_security.openssl_ciphers()))
+
+    openssl_ciphers = sdk_security.openssl_ciphers()
+
+    missing_openssl_ciphers = sdk_security.missing_openssl_ciphers(openssl_ciphers)
+    log.warning("\n%s OpenSSL ciphers missing from the cipher_suites module:", len(missing_openssl_ciphers))
+    log.warning("\n".join(sdk_utils.sort(list(missing_openssl_ciphers))))
+
+    possible_ciphers = set(filter(partial(is_not, None),
```
Comment (elezar) [inline frameworks/kafka/tests/test_tls.py]: If `openssl_ciphers` is a `set` and `missing_openssl_ciphers` is also a `set`, then isn't 
```python
possible_ciphers = set(map(cipher_suites.rfc_name, openssl_ciphers - missing_openssh_ciphers))
```
?
One may also want to update the variable names to indicate the difference between RFC and SSL names.

File: testing/security/cipher_suites.py (hunk 1 above)
Comment (mpereira) [reply testing/security/cipher_suites.py]: Oops, yes.

Comment (mpereira) [reply testing/security/cipher_suites.py]: https://github.com/mesosphere/dcos-commons/pull/2518/commits/e3e85bfd0521aaa4b9edcd78a6b268db20384981

File: frameworks/kafka/tests/test_tls.py (hunk 2 above)
Comment (mpereira) [reply frameworks/kafka/tests/test_tls.py]: > possible_ciphers = set(map(cipher_suites.rfc_name, openssl_ciphers - missing_openssh_ciphers))

Ah yes, that's better.

Regarding naming I was prefixing `openssl` when they were OpenSSL names and not adding any prefix when they were RFC names. Do you think we should instead prefix `rfc`?

Comment (mpereira) [reply frameworks/kafka/tests/test_tls.py]: https://github.com/mesosphere/dcos-commons/pull/2518/commits/7364f4d353b177e4be6f782a1ac1fb7d0fbb6221

Comment (elezar) [reply frameworks/kafka/tests/test_tls.py]: Hmmm ... yes, you're right. I missed the `openssl_`. That's fine then.

File: frameworks/kafka/tests/test_tls.py [hunk 3]
```diff
@@ -120,24 +120,32 @@ def test_tls_ciphers(kafka_service_tls):
...
         json=True), '').rstrip().split(','))
-    possible_ciphers = set(map(cipher_suites.rfc_name, sdk_security.openssl_ciphers()))
+
+    openssl_ciphers = sdk_security.openssl_ciphers()
+
+    missing_openssl_ciphers = sdk_security.missing_openssl_ciphers(openssl_ciphers)
+    log.warning("\n%s OpenSSL ciphers missing from the cipher_suites module:", len(missing_openssl_ciphers))
+    log.warning("\n".join(sdk_utils.sort(list(missing_openssl_ciphers))))
+
+    possible_ciphers = set(map(cipher_suites.rfc_name, openssl_ciphers - missing_openssl_ciphers)))
```
[the remark this answers is not in the document]
Comment (mpereira) [reply frameworks/kafka/tests/test_tls.py]: That's true. I'll push the change.

I think I'll keep the [logging of the RFC names](https://github.com/mesosphere/dcos-commons/pull/2518/files/7364f4d353b177e4be6f782a1ac1fb7d0fbb6221#diff-02c6a72f5c6bdb949062db41fed380e5R140), maybe improve it to also contain the OpenSSL names too, so we'd still need the other mapping direction.

[the remark this answers is not in the document]
Comment (mpereira) [reply frameworks/kafka/tests/test_tls.py]: https://github.com/mesosphere/dcos-commons/pull/2518/commits/589134f7da0770e7292d2180c1c9ef45d019cefb

**Comment by mpereira** [comment]
Thanks @elezar. Fwiw, we're still logging roughly the same amount of lines as before.

**Comment by nickbp** [comment]
Retriggered CI: This PR was affected by an outage of `json-schema.org` in one of our dependencies. That's since been cleaned up so we should be good to go.

File: frameworks/kafka/tests/test_tls.py [hunk 4]
```diff
@@ -120,24 +120,30 @@ def test_tls_ciphers(kafka_service_tls):
...
+    log.info("\n%s ciphers will be checked:", len(possible_openssl_ciphers))
+    for openssl_cipher in sdk_utils.sort(possible_openssl_ciphers):
+        log.info("%s (%s)", cipher_suites.rfc_name(openssl_cipher), openssl_cipher)
 
-    for cipher in possible_ciphers:
-        openssl_cipher = cipher_suites.openssl_name(cipher)
+    for openssl_cipher in possible_openssl_ciphers:
         if sdk_security.is_cipher_enabled(config.SERVICE_NAME, task_name, openssl_cipher, endpoint):
-            enabled_ciphers.add(cipher)
+            enabled_ciphers.add(sdk_security.rfc_name(openssl_cipher))
```
Comment (elezar) [inline frameworks/kafka/tests/test_tls.py]: @mpereira same `sdk_security` -> `cipher_suites` issue?

Comment (mpereira) [reply frameworks/kafka/tests/test_tls.py]: Ugh, yes.

**Comment by mpereira** [comment]
HW failures are unrelated to the PR, merging.

**Pull request merged**